webpack/scenes/ContentViews/Table/ContentViewsTable.js

```diff
@@ -0,0 +1,132 @@
+
+const ContentViewTable = ({
+  loadContentViewDetails, detailsMap, results, loading,
+}) => {
+  const [table, setTable] = useState({ rows: [], columns: [] });
+  // Map of CV id to expanded cell, if id not present, row is not expanded
+  const [expandedColumnMap, setExpandedColumnMap] = useState({});
+  const cvsPresent = results && results.length > 0;
+
+  useEffect(
+    () => {
+      if (!loading && cvsPresent) {
+        const tableData = tableDataGenerator(
+          results,
+          detailsMap,
+          expandedColumnMap,
+        );
+        setTable(tableData);
+      }
+    },
+    [results, detailsMap, expandedColumnMap],
+  );
+
+  const cvIdFromRow = ({ details: { props: rowProps } }) => rowProps.contentviewid;
+
+  const loadDetails = (id) => {
+    if (detailsMap[id]) return;
+    loadContentViewDetails(id);
+  };
+
+  const onSelect = (event, isSelected, rowId) => {
+    let rows;
+    if (rowId === -1) {
+      rows = table.rows.map(row => ({ ...row, selected: isSelected }));
+    } else {
+      rows = [...table.rows];
+      rows[rowId].selected = isSelected;
+    }
+
+    setTable(prevTable => ({ ...prevTable, rows }));
+  };
+
+  const onExpand = (_event, _rowIndex, colIndex, isOpen, rowData) => {
+    const { rows } = table;
+    const contentViewId = cvIdFromRow(rowData);
+    // adjust for the selection checkbox cell being counted in the index
+    const adjustedColIndex = colIndex - 1;
+
+    if (!isOpen) {
+      setExpandedColumnMap(prev => ({ ...prev, [contentViewId]: adjustedColIndex }));
+    } else {
+      // remove the row completely by assigning it to a throwaway variable
+      // eslint-disable-next-line camelcase, no-unused-vars
+      const { [contentViewId]: _throwaway, ...newMap } = expandedColumnMap;
+      setExpandedColumnMap(newMap);
+    }
+    loadDetails(contentViewId);
+
+    setTable(prevTable => ({ ...prevTable, rows }));
+  };
+
+  const actionResolver = (rowData, { _rowIndex }) => {
+    // don't show actions for the expanded parts
+    if (rowData.parent || rowData.compoundParent || rowData.noactions) return null;
+
+    // printing to the console for now until these are hooked up
+    /* eslint-disable no-console */
+    return [
+      {
+        title: 'Publish and Promote',
+        onClick: (_event, rowId, rowInfo) => console.log(`clicked on row ${rowId} with Content View ${cvIdFromRow(rowInfo)}`),
+      },
+      {
+        title: 'Promote',
+        onClick: (_event, rowId, rowInfo) => console.log(`clicked on row ${rowId} with Content View ${cvIdFromRow(rowInfo)}`),
+      },
+      {
+        title: 'Copy',
+        onClick: (_event, rowId, rowInfo) => console.log(`clicked on row ${rowId} with Content View ${cvIdFromRow(rowInfo)}`),
+      },
+      {
+        title: 'Delete',
+        onClick: (_event, rowId, rowInfo) => console.log(`clicked on row ${rowId} with Content View ${cvIdFromRow(rowInfo)}`),
+      },
+    ];
+    /* eslint-enable no-console */
```

Comment:
maybe instead of using the console here at the moment,
we can create the redux actions for those, so the infrastructure will already exist,
and we could see the action in Redux.

Reply:
IMHO It feels a bit early and out of scope for this PR for that. This will definitely be refactored (probably abstracted out of the component) and the actions will be added to redux in a future PR as this page is built out more, where we can discuss more about the structure.

Reply:
I agree these should eventually be in Redux but it's not needed this early